src/workload.py

```diff
@@ -64,8 +66,38 @@ def alive(self) -> bool:
 
     @property
     @override
+    @retry(
+        wait=wait_fixed(2),
+        stop=stop_after_attempt(5),
+        retry=retry_if_not_result(lambda result: True if result else False),
+        retry_error_callback=(lambda state: state.outcome.result()),  # type: ignore
+    )
     def healthy(self) -> bool:
-        return self.container.get_service(self.container.name).is_running()
+        """Flag to check if the unit service is reachable and serving requests."""
```

Comment:
**question:** Having this, is the health check on the pebble layer still relevant?

Reply:
I think so, as the pebble layer check is ran on the charm container, whereas these events won't run if the workload container is inactive.